cylc/flow/network/client.py

```diff
@@ -374,6 +385,7 @@ def get_header(self) -> dict:
                     os.getenv(
                         "CLIENT_COMMS_METH",
                         default=CommsMeth.ZMQ.value
-                    )
+                    ),
+                'behalf_of': behalf_of
```

Comment:
Send the full message through using the existing `auth_user` field rather than creating a new `behalf_of` one.

That way no changes will be required in `cylc/flow/network/resolvers.py` and the feature will be backwards compatible.

Reply:
ZMQ doesn't come into it, just replace `behalf_of` with `auth_user` and supply it with the full text:

```suggestion
                'auth_user': f'{actor} (possibly on behalf of {user})'
```